src/Entity/PrintArticleTypeInterface.php

```diff
@@ -33,4 +33,28 @@ public function getGrid();
    */
   public function getIdms();
 
+  /**
+   * Returns the thumbnail file entity.
+   *
+   * @return \Drupal\file\FileInterface|bool
+   *   The thumbnail's file entity or FALSE if thumbnail does not exist.
+   */
+  public function getThumbnailFile();
+
+  /**
+   * Returns the thumbnail's URL.
+   *
+   * @return string|bool
+   *   The thumbnail's URL or FALSE if icon does not exits.
+   */
+  public function getThumbnailUrl();
+
+  /**
+   * Extract thumbnail from idms and create file entity.
+   *
+   * @return \Drupal\file\FileInterface|bool
+   *   The thumbnail's file entity or FALSE if thumbnail does not exist.
+   */
+  public function retrieveThumbnail();
```

Comment:
@chrfritsch I would call it "buildThumbnail()" as this is everything it does. Besides that I'm a huge fan of not return two different data types. So returning `NULL` instead of `FALSE` would be better here IMHO.